app/models/team.rb

```diff
@@ -18,6 +18,39 @@ def to_s
     name
   end
 
+  def user_deletion_history
+    PaperTrail::Version.
+      where(event: 'destroy', item_type: 'UserTeam').
+      where("object ->>'group_id' = '?'", id)
+  end
+
+  def user_deletion_report_item(deleted_record)
+    {
+      user_id: deleted_record['user_id'],
+      user_email: User.find_by(id: deleted_record['user_id'])&.email,
+      team_id: deleted_record['group_id'],
+      team_name: Team.find_by(id: deleted_record['group_id'])&.name,
+      removed_at: deleted_record['removed_at'],
+      whodunnit_id: deleted_record['whodunnit_id'],
+      whodunnit_email: User.find_by(id: deleted_record['whodunnit_id'])&.email,
+    }
+  end
+
+  def user_deletion_history_report(email = nil)
+    user_deletion_history.
+      order(created_at: :desc).
+      limit(5000).
```

Comment:
let's make limit an option here too

```suggestion
  def user_deletion_history_report(email: nil, limit: 5000)
    user_deletion_history.
      order(created_at: :desc).
      limit(limit).
```